src/app/containers/PageHandlers/withData.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { shape } from 'prop-types';
+import articlePropTypes from '../../models/propTypes/article';
+import ErrorMain from '../ErrorMain';
+import nodeLogger from '../../helpers/logger.node';
+
+const logger = nodeLogger(__filename);
+
+const WithData = Component => ({ data }) => {
+  let status;
+
+  if (data) {
+    const { data: articleData } = data;
+    let { status } = data;
+
+    if (status && articleData && status === 200) {
+      try {
+        return <Component articleData={articleData} />;
+      } catch (err) {
+        logger.error(err);
+      }
+    }
+  }
+
+  return <ErrorMain status={status || 500} />;
```

Comment:
Would we want to be defining the default status code in ErrorMain itself? There is already a case of this in ErrorMain itself: https://github.com/bbc/simorgh/blob/37ca772ed0f6a8630693578fcd3fc4814e90fe79/src/app/containers/ErrorMain/index.jsx#L35

Reply:
Im personally a fan of being explicit of the type of error we should show, rather than relying on the defaults, however I see your point Ross :)